API/pubsub/README.md

```diff
@@ -0,0 +1,72 @@
+pubsub API
+==========
+
+#### `pubsub.subscribe`
+
+> Subscribe to a pubsub topic.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.pubsub.subscribe(topic, options, handler, callback)
+
+- `topic: string`
+- `options: Object` - (Optional), might contain the following properties:
+  - `discover`: type: Boolean - Will use the DHT to find
+- `handler: (msg) => ()` - Event handler which will be called with a message object everytime one is received. The `msg` has the format `{from: string, seqno: Buffer, data: Buffer, topicCIDs: Array<string>}`.
+- `callback: (Error) => ()` (Optional) Called once the subscription is established.
+
+If no `callback` is passed, a [promise][] is returned.
+
+> _In the future, topic can also be type of TopicDescriptor (https://github.com/libp2p/pubsub-notes/blob/master/flooding/flooding.proto#L23). However, for now, only strings are supported._
+
+#### `pubsub.unsubscribe`
+
+> Unsubscribes from a pubsub topic.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - `ipfs.pubsub.unsubscribe(topic, handler)`
+
+- `topic: string` - The topic to unsubscribe from
+- `handler: (msg) => ()` - The handler to remove.
+
+This works like `EventEmitter.removeListener`, as that only the `handler` passed to a `subscribe` call before is removed from listening. The underlying subscription will only be canceled once all listeners for a topic have been removed.
+
+#### `pubsub.publish`
+
+> Publish a data message to a pubsub topic.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.pubsub.publish(topic, data, callback)
+
+- `topic: string`
+- `callback: (Error) => ()` - Calls back with an error or nothing if the publish was successfull.
+
+- Returns: `Promise` if no `callback` was passed otherwise `undefined`.
```

Comment:
We have used always: `If no `callback` is passed, a [promise][] is returned.`, I'm fine changing the sentence, but I do like consistency :)